File: pinyougou/pinyougou-seckill/pinyougou-seckill-interface/src/main/java/com/pinyougou/seckill/service/SeckillGoodsService.java
package com.pinyougou.seckill.service;

import com.pinyougou.pojo.TbSeckillGoods;
import com.pinyougou.service.BaseService;
import com.pinyougou.vo.PageResult;

import java.util.List;

public interface SeckillGoodsService extends BaseService<TbSeckillGoods> {

    PageResult search(Integer page, Integer rows, TbSeckillGoods seckillGoods);

    /**
     * 查询可以秒杀的商品列表
     * @return 秒杀商品列表
     */
    List<TbSeckillGoods> findList();

    /**
     * 根据秒杀商品id从redis中获取该商品
     * @param id 秒杀商品id
     * @return 秒杀商品
     */
    TbSeckillGoods findOneInRedisBySeckillId(Long id);
}

File: pinyougou/pinyougou-seckill-web/src/main/webapp/js/service/seckillGoodsService.js
//定义业务服务
app.service("seckillGoodsService",function ($http) {

    this.findList = function () {
        return $http.get("../seckillGoods/findList.do");
    };

    this.findOne = function (id) {
        return $http.get("../seckillGoods/findOne.do?id=" + id);
    };

    this.submitOrder = function (id) {
        return $http.get("../seckillOrder/submitOrder.do?seckillId=" + id);
    };
});